### ISSUE TITLE
TPPOVM gpindices not set properly through pickling

### ISSUE THREAD

**Issue by sserita** [pyGSTio/pyGSTi#324]
**Describe the bug**
The gpindices for TPPOVMs are not set properly through pickling. This appears to be a TPPOVM specific problem - states, operations, and UnconstrainedPOVMs are fine.
This is particularly insidious because `mpi4py` inherently uses pickling for interprocess communication, so fixing this is important for MPI parallelism. 

**To Reproduce**
```
from pygsti.modelpacks import smq1Q_XY

target_model_1q=smq1Q_XY.target_model('full TP')

print('target_model_1q pre-pickle', flush=True)
target_model_1q._print_gpindices()

with open('model_pickle.pkl', 'wb') as f:
    pickle.dump(target_model_1q,f)
with open('model_pickle.pkl', 'rb') as f:    
    target_model_1q_post = pickle.load(f)

print('target_model_1q post-unpickle', flush=True)
target_model_1q_post._print_gpindices()
```

The output is then:
```
target_model_1q pre-pickle
PRINTING MODEL GPINDICES!!!
>>> rho0 [TPState]: 3 params, indices=slice(0, 3, None)
>>> Mdefault [TPPOVM]: 4 params, indices=slice(3, 7, None)
  >>> Sub0 [FullPOVMEffect]: 4 params, indices=slice(3, 7, None)
    >>> Sub0 [FullState]: 4 params, indices=slice(3, 7, None)
  >>> Sub1 [ComplementPOVMEffect]: 4 params, indices=slice(3, 7, None)
    >>> Sub0 [FullPOVMEffect]: 4 params, indices=slice(3, 7, None)
      >>> Sub0 [FullState]: 4 params, indices=slice(3, 7, None)
>>> Gxpi2:0 [FullTPOp]: 12 params, indices=slice(7, 19, None)
>>> Gypi2:0 [FullTPOp]: 12 params, indices=slice(19, 31, None)
target_model_1q post-unpickle
PRINTING MODEL GPINDICES!!!
>>> rho0 [TPState]: 3 params, indices=slice(0, 3, None)
>>> Mdefault [TPPOVM]: 4 params, indices=slice(3, 7, None)
  >>> Sub0 [FullPOVMEffect]: 4 params, indices=slice(0, 4, None)
    >>> Sub0 [FullState]: 4 params, indices=slice(0, 4, None)
  >>> Sub1 [ComplementPOVMEffect]: 4 params, indices=slice(0, 4, None)
    >>> Sub0 [FullPOVMEffect]: 4 params, indices=slice(0, 4, None)
      >>> Sub0 [FullState]: 4 params, indices=slice(0, 4, None)
>>> Gxpi2:0 [FullTPOp]: 12 params, indices=slice(7, 19, None)
>>> Gypi2:0 [FullTPOp]: 12 params, indices=slice(19, 31, None)
```

**Expected behavior**
It is expected that the gpindices are unchanged through pickling, i.e. the line "  >>> Sub0 [FullPOVMEffect]: 4 params, indices=slice(0, 4, None)" should be "  >>> Sub0 [FullPOVMEffect]: 4 params, indices=slice(3, 7, None)".

**Environment (please complete the following information):**
 - pyGSTi v0.9.11.1
 - python 3.10
 - OSX 13.4

**Comment by sserita** [comment]
Being fixed in #336

**Comment by sserita** [comment]
Closing with the release of 0.9.11.2
